scripts/stats-aggregator/overall.ts

```diff
@@ -0,0 +1,246 @@
+
+import { BlockNumberRange, prisma } from "@blobscan/db";
+
+import { deleteOptionDef, helpOptionDef } from "./common";
+
+const BEACON_NODE_ENDPOINT = process.env.BEACON_NODE_ENDPOINT;
+const DEFAULT_UNPROCESSED_BLOCKS_BATCH_SIZE = 100_000;
+
+type BeaconFinalizedBlockResponse = {
+  data: {
+    message: {
+      slot: string;
+      body: {
+        execution_payload: {
+          block_number: string;
+        };
+      };
+    };
+  };
+};
+
+type Block = {
+  number: number;
+  slot: number;
+};
+
+type BlockId = number | "latest" | "finalized";
+
+const overallCommandOptDefs: commandLineUsage.OptionDefinition[] = [
+  deleteOptionDef,
+  helpOptionDef,
+  {
+    name: "to",
+    alias: "t",
+    typeLabel: "{underline block-id}",
+    description: `Block identifier up to which to aggregate data. It can be a block number, "latest" for the last indexed block or "finalized"  for the chain's most recent finalized block. It defaults to "finalized"`,
+    defaultValue: "finalized",
+    type: String,
+  },
+  {
+    name: "batchSize",
+    alias: "s",
+    typeLabel: "{underline size}",
+    description: `Number of blocks to process in a single batch. It defaults to ${DEFAULT_UNPROCESSED_BLOCKS_BATCH_SIZE}`,
+    defaultValue: DEFAULT_UNPROCESSED_BLOCKS_BATCH_SIZE,
+    type: Number,
+  },
+];
+
+const overallCommandUsage = commandLineUsage([
+  {
+    header: "Overall Command",
+    content: "Aggregate overall stats.",
+  },
+  {
+    header: "Options",
+    optionList: overallCommandOptDefs,
+  },
+]);
+
+async function getBlockFromBeacon(id: BlockId): Promise<Block> {
+  let response: Response;
+
+  try {
+    response = await fetch(
+      `${BEACON_NODE_ENDPOINT}/eth/v2/beacon/blocks/${id}`
+    );
+  } catch (err) {
+    const err_ = err as Error;
+
+    throw new Error(
+      `Failed to fetch block from beacon node: ${err_.cause ?? err_.message}`,
+      {
+        cause: err_.cause,
+      }
+    );
+  }
+
+  const {
+    data: { message },
+  } = (await response.json()) as BeaconFinalizedBlockResponse;
+
+  return {
+    number: Number(message.body.execution_payload.block_number),
+    slot: Number(message.slot),
+  };
+}
+
+export async function deleteOverallStats() {
+  await prisma.$transaction([
+    prisma.blobOverallStats.deleteMany(),
+    prisma.blockOverallStats.deleteMany(),
+    prisma.transactionOverallStats.deleteMany(),
+    prisma.blockchainSyncState.upsert({
+      create: {
+        lastFinalizedBlock: 0,
+        lastSlot: 0,
+      },
+      update: {
+        lastFinalizedBlock: 0,
+      },
+      where: {
+        id: 1,
+      },
+    }),
+  ]);
+
+  console.log("Overall stats delete operation executed: All stats deleted.");
+}
+
+export async function incrementOverallStats({
+  targetBlockId,
+  batchSize = DEFAULT_UNPROCESSED_BLOCKS_BATCH_SIZE,
+}: {
+  targetBlockId: BlockId;
+  batchSize?: number;
+}) {
+  if (!!BEACON_NODE_ENDPOINT && targetBlockId === "finalized") {
```

Comment:
The condition here is not doing what is supposed when `BEACON_NODE_ENDPOINT` is defined and the `targetBlockId === "finalized". We enter the condition anyways.